L1Trigger/L1TMuonOverlapPhase2/src/InputMakerPhase2.cc

```diff
@@ -0,0 +1,187 @@
+      addDTphiDigi(muonStubsInLayers, digiIt, dtThDigis.product(), iProcessor, procTyp);
+
+      auto& dtP2Digi = procDataTree.add_child("dtP2Digi", boost::property_tree::ptree());
+      dtP2Digi.add("<xmlattr>.whNum", digiIt.whNum());
+      dtP2Digi.add("<xmlattr>.scNum", digiIt.scNum());
+      dtP2Digi.add("<xmlattr>.stNum", digiIt.stNum());
+      dtP2Digi.add("<xmlattr>.slNum", digiIt.slNum());
+      dtP2Digi.add("<xmlattr>.quality", digiIt.quality());
+      dtP2Digi.add("<xmlattr>.rpcFlag", digiIt.rpcFlag());
+      dtP2Digi.add("<xmlattr>.phi", digiIt.phi());
+      dtP2Digi.add("<xmlattr>.phiBend", digiIt.phiBend());
+    }
+  }
+
+  if (!mergePhiAndTheta) {
+    for (auto& thetaDigi : (*(dtThDigis->getContainer()))) {
+      if (thetaDigi.bxNum() >= bxFrom && thetaDigi.bxNum() <= bxTo) {
+        addDTetaStubs(muonStubsInLayers, thetaDigi, iProcessor, procTyp);
+      }
+    }
+  }
+  //std::cout<<__FUNCTION__<<":"<<__LINE__<<" iProcessor "<<iProcessor<<std::endl;
+  //angleConverter->AngleConverterBase::getGlobalEta(dtThDigis, 0, 0);std::endl;
+  //angleConverter->AngleConverterBase::getGlobalEta(dtThDigis, 0, 0);
+
+  for (auto& obs : observers)
+    obs->addProcesorData("linkData", procDataTree);
+}
+
+//dtThDigis is provided as argument, because in the OMTF implementation the phi and eta digis are merged (even thought it is artificial)
+void DtPhase2DigiToStubsConverterOmtf::addDTphiDigi(MuonStubPtrs2D& muonStubsInLayers,
+                                                    const L1Phase2MuDTPhDigi& digi,
+                                                    const L1MuDTChambThContainer* dtThDigis,
+                                                    unsigned int iProcessor,
+                                                    l1t::tftype procTyp) {
+  DTChamberId detid(digi.whNum(), digi.stNum(), digi.scNum() + 1);
+
+  MuonStub stub;
+
+  //converting the quality to the same encoding as in phase-1, as it is important for extrapolation
+  if (digi.quality() >= 6)
+    stub.qualityHw = digi.quality() - 2;
+  else if (digi.quality() >= 3) {
+    if (digi.slNum() == 3)
+      stub.qualityHw = 3;
+    else if (digi.slNum() == 1)
+      stub.qualityHw = 2;
+  } else {
+    if (digi.slNum() == 3)
+      stub.qualityHw = 1;
+    else if (digi.slNum() == 1)
+      stub.qualityHw = 0;
+  }
+
+  if (stub.qualityHw < config->getMinDtPhiQuality())
+    return;
+
+  unsigned int hwNumber = config->getLayerNumber(detid.rawId());
+  if (config->getHwToLogicLayer().find(hwNumber) == config->getHwToLogicLayer().end())
+    return;
+
+  auto iter = config->getHwToLogicLayer().find(hwNumber);
+  unsigned int iLayer = iter->second;
+  unsigned int iInput = OMTFinputMaker::getInputNumber(config, detid.rawId(), iProcessor, procTyp);
+  //MuonStub& stub = muonStubsInLayers[iLayer][iInput];
+
+  stub.type = MuonStub::DT_PHI_ETA;
+
+  //std::cout<<__FUNCTION__<<":"<<__LINE__<<" iProcessor "<<iProcessor<<std::endl;
```

Comment:
Please remove this and other commented code that does not serve to illustrate a point about in-use code.